packages/foam-vscode/static/dataviz/graph.js

```diff
@@ -93,40 +98,10 @@ function update(patch) {
 }
 
 const Actions = {
-  refresh: graphInfo =>
+  refreshWorkspaceData: graphInfo =>
     update(m => {
-      m.nodeInfo = graphInfo.nodes;
-      const links = graphInfo.links;
-
-      // compute graph delta, for smooth transitions we need to mutate objects in-place
-      const nodeIdsToAdd = new Set(Object.keys(m.nodeInfo));
-      const nodeIndexesToRemove = new Set();
-      m.data.nodes.forEach((node, index) => {
-        if (nodeIdsToAdd.has(node.id)) {
-          nodeIdsToAdd.delete(node.id);
-        } else {
-          nodeIndexesToRemove.add(index);
-        }
-      });
-      // apply the delta
-      nodeIndexesToRemove.forEach(index => {
-        m.data.nodes.splice(index, 1); // delete the element
-      });
-      nodeIdsToAdd.forEach(nodeId => {
-        m.data.nodes.push({
-          id: nodeId,
-        });
-      });
-      m.data.links = links; // links can be swapped out without problem
-
-      // check that selected/hovered nodes are still valid (see #397)
-      m.hoverNode = m.nodeInfo[m.hoverNode] != null ? m.hoverNode : null;
-      m.selectedNodes = new Set(
-        Array.from(m.selectedNodes).filter(nId => m.nodeInfo[nId] != null)
-      );
-
-      // annoying we need to call this function, but I haven't found a good workaround
-      graph.graphData(m.data);
+      m.fullGraphData = graphInfo;
+      //patchGraphData(m,graphInfo);
```

Comment:
I think by calling this function here you won't need to call `filterByType` in `didUpdateGraphData` - what was your thinking in this regard?

Reply:
I'm fine either way. I prefer not to show every node in the graph on load. So I was filtering by default.